### PR TITLE
Upgrade STM32CubeMX version from 5.3.0 -> 6.9.2

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Summary

Upgrade the boards in our repo from STM32CubeMX 5.3.0 to 6.9.2. We need to switch to 6.9.2 to unlock support for the STM32H7xx family of microcontrollers, which is going to be on our vehicle controller this year. Also, we should just upgrade since they probably fixed a bunch of bugs.

TODO: Test this on other people's machines

### Changelist 

- Migrate every board's `.ioc` file from 5.3.0 to 6.9.2. This means we had to regenerate the drivers/middleware for every board, leading to a crazy diff.
- Switch to CMSIS-RTOS v2, which is supported in 6.9.2 (it wasn't in 5.3.0). CMSIS-RTOS is an API that wraps FreeRTOS, and v2 is way easier to use than regular FreeRTOS and way more powerful than v1. 
- Bumped the version in the container (using a custom tag currently so `ubcformulaelectric/environment` still points to the regular container, and other CI jobs can run).

### Testing Done

- [x] Flashed all boards and spun wheels on jacks
- [ ] This is a big change, so lets do some driving with these changes

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: .github/workflows/bin-dbc-gen.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
 
 jobs:
   build:
-    uses: UBCFormulaElectric/Consolidated-Firmware/.github/workflows/build.yml@master
```
Comment (Lucien950) [inline .github/workflows/bin-dbc-gen.yml:9]: this is going to be fixed in peter/build_DBC_PR, hopefully no conflicts.

File: environment/Dockerfile [hunk 2]
```diff
@@ -14,27 +14,27 @@ FROM ubuntu:22.04
 # openocd: Our embedded GDB server of choice
 # usbutils: USB tools (lsusb)
 # nano: Basic Unix text editor
+# libgtk-3-0: Random library required for running STM32CubeMX in container
+# libxi6: Random library required for running STM32CubeMX V6 in container
+# libxtst6: Random library required for running STM32CubeMX V6 in container
+# libxrender1: Random library required for running STM32CubeMX V6 in container
 RUN apt update -y && apt install -y \
```
Comment (Lucien950) [inline environment/Dockerfile:21]: remove comments for unused dependencies

File: environment/Dockerfile [hunk 3]
```diff
@@ -14,27 +14,27 @@ FROM ubuntu:22.04
...
     wget \
     unzip \
     pip \
     git \
     git-lfs \
-    openjdk-11-jdk \
     xvfb \
     libncurses5 \
     valgrind \
     openocd \
```
Comment (Lucien950) [inline environment/Dockerfile:30]: i don't think we need this if we're not flashing?

File: environment/Dockerfile [hunk 4]
```diff
@@ -14,27 +14,27 @@ FROM ubuntu:22.04
...
+# libgtk-3-0: Random library required for running STM32CubeMX in container
+# libxi6: Random library required for running STM32CubeMX V6 in container
+# libxtst6: Random library required for running STM32CubeMX V6 in container
+# libxrender1: Random library required for running STM32CubeMX V6 in container
 RUN apt update -y && apt install -y \
     wget \
     unzip \
     pip \
     git \
     git-lfs \
```
Comment (Lucien950) [inline environment/Dockerfile:26]: if we clone from outside and volume, perhaps this is not used either. (git, gitlfs, unzip, wget??)

**Review by Lucien950 — CHANGES_REQUESTED** [review]
just a few optimization things.

File: environment/Dockerfile (hunk 4 above)
Comment (gtaharaedmonds) [reply environment/Dockerfile:26]: if I remember correctly I did have to install these to get the CI working but that was a long time ago
container takes too long to build for me to want to try

**Comment by gtaharaedmonds** [comment]
> Also I am not familiar about autogenerated STM files but can they autogenerate on each build? (If so can we hide it away in build_arm)

Possibly yes. Currently we invoke CubeMX with the `generate code` command which just updates `main.c` and a few other files. We can also invoke with the `project generate` which should also update the driver files. Although Cube usually doesn't like being generated in a different directory than the `.ioc` so I'm not sure if we can put it in `build_arm`. 

This'll be easier after the refactor since all CubeMX stuff will live in its own folder.

**Comment by Lucien950** [comment]
I'll review once #1040 gets merged in.

**Review by Lucien950 — APPROVED** [review]

**Pull request merged**